src/yadism/esf/tmc.py

```diff
@@ -479,7 +537,67 @@ def _get_result_exact(self):
         return self._factor_shifted * F3out + self._factor_h3 * h3out
 
 
-ESFTMCmap = {"F2": ESFTMC_F2, "FL": ESFTMC_FL, "F3": ESFTMC_F3}
+class ESFTMC_g1(EvaluatedStructureFunctionTMC):
+    """
+    This function implements the actual formula for target mass corrections
+    for parity conserving polarized structure function g1, for the two kinds
+    described in the parent class :py:class:`EvaluatedStructureFunctionTMC`.
+
+    The formula in question can be found in :cite:`Accardi:2008pc`,
+    :cite:`Khanpour:2017cha`, and references therein.
+
+    Parameters
+    ----------
+    SF : StructureFunction
+        the interface object representing the structure function kind he
+        belongs to
+    kinematics : dict
+        requested kinematic point
+
+    """
+
+    def __init__(self, SF, kinematics):
+        super().__init__(SF, kinematics)
+        # Kinematic factor for ZM g1
+        self._factor_shifted = self.x**2 / (self.xi * self.rho**3)
+        # Kinematic factor for common for `k1` & `k2` integral
+        self._factor_k1_k2 = (4 * self.x**2 * self.mu) / self.rho**3
+        # Kinematic factor specific for `k1` integral
+        self._factor_k1 = (self.x + self.xi) / self.xi
+        # Kinematic factor specific for `k2` integral
+        self._factor_k2 = (self.rho**2 - 3) / (2 * self.rho)
```

Comment:
Looking at Eq. 10 of `:cite:tmc-g1-iranian` I can somehow identify `_factor_k1`, but the 3 inside `_factor_k2` I can not identify ...

Reply:
Perhaps it might be easier to refer to Eq. 26 of `tmc-g1-accardi` as the expression there is more compact. From there, the $3$ stems from replacing $\gamma^2 -  2 \longrightarrow ((1 + \gamma^2 \to \rho^2) - 3)$.